scripts/install.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+
+BINARY_NAME="universal-sierra-compiler"
+LOCAL_BIN="${HOME}/.local/bin"
+
+main () {
+  download_and_extract_binary
+
+  addBinaryToPath
+
+  echo "${BINARY_NAME} (${release_tag}) has been installed successfully."
+}
+
+download_and_extract_binary() {
+  repo="https://github.com/software-mansion/universal-sierra-compiler"
+  # Fetch the latest release tag from GitHub API
+  release_tag=$(curl -# -Ls -H 'Accept: application/json' "${repo}/releases/latest" | sed -e 's/.*"tag_name":"\([^"]*\)".*/\1/')
+
+  # Define the operating system and architecture
+  get_architecture
+  _arch="$RETVAL"
+
+  artifact_name=${BINARY_NAME}-${release_tag}-${_arch}
+
+  echo "Downloading and extracting ${artifact_name}..."
+  # Create a temporary directory
+  temp_dir=$(mktemp -d)
+
+  # Download and extract the archive
+  curl -L "${repo}/releases/download/${release_tag}/${artifact_name}.tar.gz" | tar -xz -C "${temp_dir}"
```

Comment:
to make script more robust, let's add a check for curl and tar as some envs may not have them.

Reply:
Moreover, windows uses 7z, so we should probably factor this in as well

Reply:
I wonder if we should care about windows. I mean, we have step-by-step instruction for Windows users in the SNFoundry but snfoundryup won't work on windows

Reply:
Sure, I see it's already removed so that's good :+1:
But still, let's add a check for `curl` and `tar`, as they can be unavailable in some systems out of the box

Reply:
https://github.com/software-mansion/universal-sierra-compiler/pull/8/commits/cd5f5fff581a054ce8ae82eed0504e792fdd4157